src/presentation/controllers/indentity.ts

```diff
@@ -0,0 +1,67 @@
+import { CookieOptions, Response, Request } from 'express';
+import { httpConfig } from '../config';
+import { createSuccessResponse } from '../api-response';
+import {
+  type LoginLocalResponseDto,
+  loginSchema,
+  registerSchema,
+} from '../validators/identity-schema';
+import { LoginLocal } from '../../application/use-cases/LoginLocal';
+import { RegisterAndLoginLocal } from '../../application/use-cases/RegisterAndLoginLocal';
+
+interface IdentityRouterDeps {
+  loginLocal: Pick<LoginLocal, 'execute'>;
+  registerAndLoginLocal: Pick<RegisterAndLoginLocal, 'execute'>;
+}
+
+export class IdentityController {
+  private readonly loginLocal: Pick<LoginLocal, 'execute'>;
+  private readonly registerAndLoginLocal: Pick<RegisterAndLoginLocal, 'execute'>;
+
+  constructor({ loginLocal, registerAndLoginLocal }: IdentityRouterDeps) {
+    this.loginLocal = loginLocal;
+    this.registerAndLoginLocal = registerAndLoginLocal;
+  }
+
+  async register(req: Request, res: Response) {
+    const body = registerSchema.parse(req.body);
+    const result = await this.registerAndLoginLocal.execute(body);
+    this.setAccessTokenCookie(res, result.accessToken.token, result.accessToken.expiresAt);
+    res.status(201).end();
+  }
+
+  async login(req: Request, res: Response) {
+    const body = loginSchema.parse(req.body);
+    const { accessToken, ...rest } = await this.loginLocal.execute(body);
+    this.setAccessTokenCookie(res, accessToken.token, new Date(Date.now() + 7 * 60 * 60 * 1000));
```

Comment:
La expiación esta hardcodeada, se podría usar la que esta en el caso de uso del token y así que la cookie y el token tengan la misma consistencia, acabando al mismo tiempo tanto en lado del cliente como en el del servidor.
```suggestion
    this.setAccessTokenCookie(res, accessToken.token, accessToken.expiresAt);
```

Reply:
si se me paso.